R/createSplitGate.R

```diff
@@ -0,0 +1,81 @@
+#' Create split gate
+#'
+#' Creates a split gate. split gates have two sectors (right and left),
+#' each with a unique gid and name.
+#'
+#' @param experimentId The ID of the experiment to which to add the gate.
+#' @param xChannel The name of the x channel to which the gate applies.
+#' @param yChannel The name of the y channel to which the gate applies.
+#' @param name The name of the gate. Each sector is named with this parameter
+#' and a split flag (L, R).
+#' @param x The x coordinate (after the channel's scale has been applied).
+#' @param y The y coordinate (after the channel's scale has been applied).
+#' @param gid Top-level group ID of the gate, used for tailoring. If this is not specified,
+#'   then a new Group ID will be created. For compound gates, "gid" refers to the
+#'   top-level GID. Each sector has a unique `model.gid` and `name` to which
+#'   populations must refer.
+#' @param gids Group IDs of each sector, assigned to `model.gid`.
+#' @param parentPopulationId ID of the parent population. Use \code{NULL} for
+#'   the "ungated" population. If specified, do not specify \code{parentPopulation}.
+#' @param parentPopulation Name of the parent population. An attempt will be made
+#'   to find the population by name. If zero or more than one population exists
+#'   with the name, an error will be thrown. If specified, do not specify
+#'   \code{parentPopulationId}.
+#' @param tailoredPerFile Whether this gate is tailored per FCS file.
+#' @param fcsFileId ID of FCS file, if tailored per file. Use \code{NULL} for
+#'   the global gate in a tailored gate group. If specified, do not specify
+#'   \code{fcsFile}.
+#' @param fcsFile Name of FCS file, if tailored per file. An attempt will be made
+#'   to find the file by name. If zero or more than one file exists with the name,
+#'   an error will be thrown. If specified, do not specify \code{fcsFileId}.
+#' @param locked Prevents modification of the gate via the web interface.
+#' @param createPopulation Automatically create corresponding population.
+#' @export
+#' @examples
+#' \dontrun{
+#' createSplitGate(experimentId, "FSC-A", "my gate")
+#' }
+createSplitGate = function(experimentId, xChannel, name, x, y,
+                               gid = generateId(), gids = replicate(2, generateId()),
+                               parentPopulationId = NULL, parentPopulation = NULL, skewable = FALSE,
+                               tailoredPerFile = FALSE, fcsFileId = NULL, fcsFile = NULL,
+                               locked = FALSE, createPopulation = TRUE) {
+
+    scales = data.frame(getScaleSets(experimentId)$scales)
+    # mysteriously, it doesn't seem to matter what you set y to be
```

Comment:
y (both the label y and the split y) are the fraction of the plot height (0 to 1, 0=bottom, 1=top). You can use 0.5 to be safe. 1000 should put the label way off the plot.

Reply:
It's funny you noticed this change, since it's flagged `Outdated`; I figured that out after a while, and found 0.95 to be a good default.